include/motor-control/core/spi.hpp

```diff
@@ -0,0 +1,16 @@
+#pragma once
+
+#include <array>
+#include <concepts>
+#include <cstdint>
+
+namespace spi {
+
+template <class Spi>
+concept SpiProtocol = requires(Spi spi_comms,
+                               const std::array<uint8_t, 5>& transmit,
```

Comment:
Depending on how general we want to be, it may be a good idea to make the `5` actually a template argument; even if we don't though we should probably make it depend on the `BUFFER_SIZE` value in the motor_control.hpp (or the other way around) so it's not possible to change them independently.

I think it's probably fine to keep it specific to a number; but if we do I think we should name this concept something like `TMC2130Spi` to make it clear it's not just a generic SPI interface, it's for this specific driver with this specific packet size.